articles/network-routing/index.md

```diff
@@ -0,0 +1,145 @@
+
+We start checking the distance of adjacent nodes to 0 (nodes 1 & 2).
+We only add a node if the distance between node 0 and the node is the shortest.
+From node 0 to node 1, the distance is 2, while from 0 to 2, the distance is 6. The distance to node 1 is the shortest, so we add node 1 to the path.</br>
+*(We will mark added nodes by adding an asterisk beside them)*
+
+![Graph3](/engineering-education/network-routing/routing4.png)
+
+We then mark node 1 as visited and add it to our list. We add node 2 for we know its distance, but we will mark it since it is not yet visited.
+
+|NODE|DISTANCE|
+---|---|
+0|0 *|
+1|2 *|
+2|6|
+3|inf|
+4|inf|
+5|inf|
+6|inf|
+
+**{0,1}**
+
+
+The next nodes are node 2 and node 3. 
+The distance 0-1-3 is (2+5) = 7.
+The distance between 0-2 is 6.</br>
+We will choose a node with the shortest distance from node 0, which is node 2.</br>
+
+![Graph4](/engineering-education/network-routing/routing6.png)
+
+|NODE|DISTANCE|
+---|---|
+0|0 *|
+1|2 *|
+2|6 *|
+3|inf|
+4|inf|
+5|inf|
+6|inf|
+
+ This will continue till all the nodes are marked as visited and added.
+
+**{0,1,2}**
+
+The final result will be this:
+
+![Graph5](/engineering-education/network-routing/routing8.png)
+
+
+|NODE|DISTANCE|
+---|---|
+0|0 *|
+1|2 *|
+2|6 *|
+3|7 *|
+4|17 *|
+5|22 *|
+6|19 *|
+
+**{0,1,2,3,4,5,6}**
+
+***Reliable flooding algorithm***
+As the name suggests, *it floods each router in a network with all other neighboring routers network state information* such as the IP addresses, costs of the network, the health of the networks, etc. using **link-state packets** of the routers.</br>
+
+ For example, take a LAN with 5 routers A, B, C, D, E.
+
+![LAN-1](/engineering-education/network-routing/LAN-1.png)
+
+Each router has its state info, which it passes to its neighbors. A will send to B. B to C & D and so on until all of them have all the states.</br>
+They can then independently apply Dijkstra's algorithm to forward packets.</br>
+That said, this is not always the case with interconnected routers as shown below where the process of passing the link state packets goes on and on without stopping hence creating a condition called **looping** e.g., A will pass its packet to B, B then forwards it to C and C passes it again to A.
+
+![LAN-2](/engineering-education/network-routing/LAN-2.png)
+
+A **unique ID** is given to each link-state packets so that this problem is solved.
+When C and B receives the packet with the unique ID from A, it (A) does not send it again to B, and B does not send it to C.</br>
+
+Applying reliable flooding can be unachievable; therefore, protocols are required to help implement it.</br>
+I will talk about one, the **OSPF**(Open Shortest Path First) protocol.</br>
+This protocol divides the wide networks into **local areas**, forming a **backbone area** which shares at least one router from the bordering local areas. These shared routers form the **border routers**.</br>
```

Comment:
Make this a subsection—H4 heading, preferably.

Reply:
Noted.